src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -1862,16 +1862,16 @@ public async Task PostAsyncExpect100Continue_SendRequest_Ok(bool send100Continue
         }
 
         [Fact]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/31220")]
         public async Task PostAsyncExpect100Continue_NonSuccessResponse_RequestBodyNotSent()
         {
             string responseContent = "no no!";
 
             await Http2LoopbackServer.CreateClientAndServerAsync(async url =>
             {
                 using (var handler = new SocketsHttpHandler())
-                using (HttpClient client = CreateHttpClient())
+                using (HttpClient client = CreateHttpClient(handler))
                 {
+                    TestHelper.EnableUnencryptedHttp2IfNecessary(handler);
```

Comment:
Just a question, why do we need this here?

Reply:
It enables **unencrypted** HTTP/2 on platforms without ALPN support (e.g. OSX) where we cannot use SSL over HTTP/2. Check [this line](https://github.com/dotnet/runtime/pull/35100/files/cb60aaa2408f8423d95d3ef03ae5b6f372a1945b#diff-4210bd74dd22b1469b8bf0307675ed37R139).

Reply:
So if I get it right, we're using here `SocketsHttpHandler` just because we need to set `Expect100ContinueTimeout`, since `HttpClientHandler` doesn't have such property.
And what we do on these 2 lines here [HttpClientHandlerTest.Http2.cs#L1874-L1875](https://github.com/dotnet/runtime/blob/cb60aaa2408f8423d95d3ef03ae5b6f372a1945b/src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs#L1874-L1875) is pretty much the same as what's done in [HttpClientHandlerTestBase.SocketsHttpHandler.cs#L22-L23](https://github.com/dotnet/runtime/blob/cb60aaa2408f8423d95d3ef03ae5b6f372a1945b/src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTestBase.SocketsHttpHandler.cs#L22-L23).

Reply:
Correct.